test/object-store/c_api/c_api.cpp

```diff
@@ -2624,6 +2664,13 @@ TEST_CASE("C API", "[c_api]") {
                     size_t size;
                     CHECK(checked(realm_list_size(bars.get(), &size)));
                     CHECK(size == 2);
+
```

Comment:
@ironage @jedelbo I noticed for all the `CollectionBase` objects we don't allow searching with, type `type_TypedLink` . Is this still wanted? Or it is something we should enable?

Reply:
A list of TypedLink is not officially supported. But there should be some way to find a destination object in a list of links. In C++ we support finding by ObjKey, but it looks like the C-API doesn't expose this type in a `realm_value_t` for some reason. We should either convert TypedLink to ObjKey in `realm_list_find` or we should support `LnkLst::find_any` given a Mixed of type TypedLink. In either case, when converting from TypedLink to ObjKey, we'd need to check that the destination table matches.

Reply:
I think probably we should file a separate issue and add support for this stuff at some stage in C API too.

Reply:
I just added a small fix, I think it is OK to sneak in here.

Reply:
OK, then we need to do this also for `Sets` and `Dictionaries`, right?

Reply:
I think we need to do nothing for `Dictionaries`.  I added the implementation for `Sets`